tests/analysis/impl/dense/gpcca_test.py

```diff
@@ -0,0 +1,468 @@
+import pytest
+import numpy as np
+
+from scipy.linalg import hilbert, pinv
+from tests.get_input import get_known_input, mu
+from tests.numeric import assert_allclose
+from msmtools.analysis.dense.gpcca import (
+    GPCCA,
+    _do_schur,
+    _fill_matrix,
+    _gram_schmidt_mod,
+    _indexsearch,
+    _initialize_rot_matrix,
+    _objective,
+    _opt_soft,
+    _cluster_by_isa,
+)
+
+eps = np.finfo(np.float64).eps * 1e10
+
+
+class TestGPCCAMatlabRegression:
+    def test_empty_P(self):
+        with pytest.raises(
+            AssertionError, match=r"Expected shape 2 but given array has shape \d+"
+        ):
+            GPCCA(np.array([]))
+
+    def test_non_square_P(self):
+        with pytest.raises(
+            AssertionError, match=r"Given array is not uniform \n\[\d+ \d+\]"
+        ):
+            GPCCA(np.random.normal(size=(4, 3)))
+
+    def test_empty_sd(self, P: np.ndarray):
+        with pytest.raises(ValueError):
+            GPCCA(P, eta=[])
+
+    def test_too_small_kkmin(self, P: np.ndarray, sd: np.ndarray):
+        g = GPCCA(P, eta=sd)
+        with pytest.raises(ValueError):
+            g.minChi(m_min=0, m_max=10)
+
+    def test_kmin_dtype(self, P: np.ndarray, sd: np.ndarray):
+        g = GPCCA(P, eta=sd)
+        with pytest.raises(TypeError):
+            g.minChi(m_min=3.0, m_max=4)
+
+    def test_kmax_dtype(self, P: np.ndarray, sd: np.ndarray):
+        g = GPCCA(P, eta=sd)
+        with pytest.raises(TypeError):
+            g.minChi(m_min=3, m_max=4.0)
+
+    def test_k_input(self, P: np.ndarray, sd: np.ndarray):
+        g = GPCCA(P, eta=sd)
+        with pytest.raises(ValueError):
+            g.minChi(m_min=5, m_max=3)
+
+    @pytest.mark.xfail
+    def test_normal_case(
+        self,
+        P: np.ndarray,
+        sd: np.ndarray,
+        count_sd: np.ndarray,
+        count_Pc: np.ndarray,
+        count_A: np.ndarray,
+        count_chi: np.ndarray,
+    ):
+        assert_allclose(sd, count_sd)
+
+        g = GPCCA(P, eta=sd)
+        g.optimize((2, 10))
+
+        Pc = g.coarse_grained_transition_matrix
+        assert_allclose(Pc, count_Pc, atol=eps)
```

Comment:
Add assertions for:
- membership vectors summing to one row-wise
- compare the identified `n_states` with ground truth
- check whether `P_c` is a transition matirx
- check whether the membership vectors are still invariant under P